src/core/connected_clients.rs

```diff
@@ -0,0 +1,45 @@
+use bevy::prelude::*;
+
+use crate::core::ClientId;
+
+#[derive(Resource, Default)]
+pub struct ConnectedClients {
```

Comment:
Maybe add `#[derive(Deref)]` and `#[deref]` for clients field?
I think it will be more convenient to have all Vec and slice methods available. And tests will look a little nicer.